File: FizzBuzzTests/fizzbuzz_test.r
# Unit tests for FizzBuzz implementation in R

library(testthat)

test_that("numbers divisible by 3 return 'Fizz'", {
  expect_equal(fizzbuzz(3), "Fizz")
  expect_equal(fizzbuzz(6), "Fizz")
})

test_that("numbers divisible by 5 return 'Buzz'", {
  expect_equal(fizzbuzz(5), "Buzz")
  expect_equal(fizzbuzz(10), "Buzz")
})

test_that("numbers divisible by both 3 and 5 return 'FizzBuzz'", {
  expect_equal(fizzbuzz(15), "FizzBuzz")
  expect_equal(fizzbuzz(30), "FizzBuzz")
})

test_that("numbers not divisible by either 3 or 5 return the number itself", {
  expect_equal(fizzbuzz(1), "1")
  expect_equal(fizzbuzz(2), "2")
})

# Main function to run the tests
test_dir("tests")

File: FizzBuzzTests/fizzbuzz_test.js
const fizzBuzz = require('../FizzBuzz/fizzbuzz.js');

describe('FizzBuzz JavaScript Tests', () => {
    test('should return "Fizz" for multiples of 3', () => {
        expect(fizzBuzz(3)).toBe('Fizz');
        expect(fizzBuzz(6)).toBe('Fizz');
    });

    test('should return "Buzz" for multiples of 5', () => {
        expect(fizzBuzz(5)).toBe('Buzz');
        expect(fizzBuzz(10)).toBe('Buzz');
    });

    test('should return "FizzBuzz" for multiples of both 3 and 5', () => {
        expect(fizzBuzz(15)).toBe('FizzBuzz');
        expect(fizzBuzz(30)).toBe('FizzBuzz');
    });

    test('should return the number for numbers not multiple of 3 or 5', () => {
        expect(fizzBuzz(1)).toBe('1');
        expect(fizzBuzz(2)).toBe('2');
    });
});
